libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -191,6 +247,100 @@ void ControlFlowAnalyzer::checkUnreachable(CFGNode const* _entry, CFGNode const*
 		// Extend the location, as long as the next location overlaps (unreachable is sorted).
 		for (; it != unreachable.end() && it->start <= location.end; ++it)
 			location.end = std::max(location.end, it->end);
-		m_errorReporter.warning(5740_error, location, "Unreachable code.");
+
+		if (m_previousUnreachable.emplace(location).second)
+			m_errorReporter.warning(5740_error, location, "Unreachable code.");
 	}
 }
+
+bool ControlFlowAnalyzer::checkForReverts(ContractDefinition const& _contract, FunctionCall const& _functionCall)
```

Comment:
Can this be done in a way where we have an isolated component that modifies the function flows to cut edges from functions where only the revert node is reachable and then run the other analyses?

Reply:
Possibly. I didn't use that solution because I thought we want a proper separation of building the graph and then analyzing it. 
have to think a bit about..

Reply:
Indeed, but I think we should have as well a separation between analyzing it and adding the fact that some functions revert.